compiler/x/codegen/X86Ops.hpp

```diff
@@ -414,6 +423,11 @@ class TR_X86OpCode
       // OpCodes beyond FENCE are not X86 instructions and hence they do not have binary encodings.
       return op < FENCE ? _binaries[op].encode<TBuffer>(cursor, rex) : cursor;
       }
+   static inline void finalize(TR_X86OpCodes op, uint8_t* cursor)
+      {
+      if (op < FENCE)
```

Comment:
This test for finalization seems too fragile, even guarded with a comment.  I wonder if you could add a "finalization" property to an opcode and peek at the appropriate `_properties` table for the given `op` to decide whether to do this?